Packs/ThreatConnect/Integrations/ThreatConnect_v2/ThreatConnect_v2.py

```diff
@@ -0,0 +1,2138 @@
+
+
+def tc_get_tags_command():
+    raw_response = tc_get_tags()
+    tags = [t['name'] for t in raw_response['data']['tag']]
+
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': raw_response,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': tableToMarkdown('ThreatConnect Tags:', tags, headers='Name'),
+        'EntryContext': {'TC.Tags': tags}
+    })
+
+
+def tc_get_tags():
+    tc = get_client()
+    ro = RequestObject()
+    ro.set_http_method('GET')
+    ro.set_request_uri('/v2/tags')
+
+    return tc.api_request(ro).json()
+
+
+def tc_tag_indicator_command():
+    args = demisto.args()
+    indicator = args['indicator']
+    tag = args['tag']
+    owners = args.get('owner')
+    indicators = tc_tag_indicator(indicator, tag, owners)
+
+    md = []
+    for ind in indicators:
+        md.append('Indicator {} with ID {}, was tagged with: {}'.format(indicator, ind.id, tag))
+
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['text'],
+        'Contents': '\n'.join(md)
+    })
+
+
+def tc_tag_indicator(indicator, tag, owners=None):
+    tc = get_client()
+    indicators = tc.indicators()
+    filter1 = indicators.add_filter()
+    filter1.add_indicator(indicator)
+
+    if owners is not None:
+        owners = owners.split(",")
+        filter1.add_owner(owners)
+
+    indicators = indicators.retrieve()
+    for indicator in indicators:
+        indicator.add_tag(tag)
+        indicator.commit()
+
+    return indicators
+
+
+def tc_get_indicator_command():
+    args = demisto.args()
+    owners = args.get('owners')
+    if not owners:
+        if 'defaultOrg' in demisto.params():
+            owners = demisto.params().get('defaultOrg')
+        else:
+            return_error('You must specify an owner in the command, or by using the Organization parameter.')
+    rating_threshold = int(args.get('ratingThreshold', -1))
+    confidence_threshold = int(args.get('confidenceThreshold', -1))
+    indicator = args['indicator']
+    associated_groups = json.loads(args['group_associations'].lower())
+    associated_indicators = json.loads(args['indicator_associations'].lower())
+    include_tags = json.loads(args['indicator_tags'].lower())
+    include_observations = json.loads(args['indicator_observations'].lower())
+    if 'indicator_type' in args:
+        indicator_type = args['indicator_type']
+    else:
+        indicator_type = None
+
+    ec, indicators, raw_indicators, indicators_associations, indicator_groups, indicator_observations, indicator_tags \
+        = tc_get_indicator(indicator, owners, rating_threshold, confidence_threshold, associated_groups,
+                           associated_indicators, include_observations, include_tags, indicator_type)
+    # remove extra items from the indicator markdown
+    if str(ec) == '[]':
+        ec = {}
+    if len(ec) > 0:
```

Comment:
`ec` is a list.
converting to string in order to check if empty. if so, convert to dict.
finally, treating it as a list.

please fix this weird code.